storage/ndb/rest-server/data-access-rondb/src/db-operations/pk/common.cpp

```diff
@@ -364,78 +345,38 @@ RS_Status SetOperationPKCol(const NdbDictionary::Column *col, NdbOperation *oper
-
-      E.g. given base64 string "YQ==" (4 ASCII characters)
-        -> data_len == 4 bytes
-        -> decoded_size == 3 bytes (since base64 always uses 4 bytes to represent 3 bytes of data)
-    */
-    size_t decoded_size = boost::beast::detail::base64::decoded_size(data_len);
-
-    if (decoded_size > size_t(col->getLength())) {
-      
-      /*
-        As described above, it may be the case that the decoded_size falsely interpretes
-        1 or 2 padding bytes as part of the data. Since the decoded_size must be a
-        multiple of 3, it may now only be the next multiple after col->getLength().
-
-        E.g.: col->getLength() == 100; decoded_size is now only allowed to be 102
-      */
-      int base64Mod = col->getLength() % 3; // 100 % 3 == 1
-      int base64Total = col->getLength() - base64Mod + 3; // 100 - 1 + 3 == 102
-
-      if (decoded_size != size_t(base64Total)) {
-        return RS_CLIENT_ERROR(
-            std::string(ERROR_008) +
-            " Decoded data length (" + std::to_string(decoded_size) +
-            ") is greater than column length (" + std::to_string(col->getLength()) +
-            "). Column: " + std::string(col->getName())
-        );
-      }
-      /*
-        It may now still be the case that the real decoded_size is 1 byte larger than
-        the column length. But we cannot find out unless we decode the string.
-      */
-    }
-
-    // operation->equal expects a zero-padded char string
-    char pk[MAX_KEY_SIZE_IN_WORDS*4];
-    std::fill (pk, pk + col->getLength(), 0);
-
-    std::pair<std::size_t, std::size_t> ret = boost::beast::detail::base64::decode(pk, encodedStr, data_len);
-
-    // ret.first does not interpret padding bytes as part of data
-    if (ret.first > size_t(col->getLength())) {
-        return RS_CLIENT_ERROR(
-            std::string(ERROR_008) +
-            " ret.first (" + std::to_string(ret.first) +
-            ") is not equal to decoded_size (" + std::to_string(decoded_size)
-        );
+    /// Binary data is sent as base64 string
+    assert(col->getLength() <= BINARY_MAX_SIZE_IN_BYTES);
+    const char *encoded_str      = request->PKValueCStr(colIdx);
+    const size_t encoded_str_len = request->PKValueLen(colIdx);
+    size_t col_len               = col->getLength();
+
+    size_t decoded_size = boost::beast::detail::base64::decoded_size(encoded_str_len);
+    if (decoded_size > BINARY_MAX_SIZE_IN_BYTES_DECODED) {
+      return RS_CLIENT_ERROR(std::string(ERROR_008) + " " +
+                             "Decoded data length is greater than column length. " +
+                             "Column: " + std::string(col->getName()) +
+                             " Length: " + std::to_string(col->getLength()));
+    }
+
+    char pk[BINARY_MAX_SIZE_IN_BYTES_DECODED];
+    std::fill(pk, pk + col->getLength(), 0);
+    std::pair<std::size_t, std::size_t> ret =
+        boost::beast::detail::base64::decode(pk, encoded_str, encoded_str_len);
```

Comment:
Not sure I am a fan of using Boost library more than necessary, pretty sure there is a Base64 function in MySQL already, otherwise I have one myself as well. Using Boost makes it impossible to optimise this function in the future.